kernel/comps/pci/src/arch/x86/mod.rs

```diff
@@ -2,23 +2,42 @@
 
 //! PCI bus access
 
-use super::device::io_port::{ReadWriteAccess, WriteOnlyAccess};
-use crate::{bus::pci::PciDeviceLocation, io::IoPort, prelude::*};
+use ostd::{
+    arch::device::io_port::{ReadWriteAccess, WriteOnlyAccess},
+    io::IoPort,
+    Error,
+};
+use spin::Once;
 
-static PCI_ADDRESS_PORT: IoPort<u32, WriteOnlyAccess> = unsafe { IoPort::new(0x0CF8) };
-static PCI_DATA_PORT: IoPort<u32, ReadWriteAccess> = unsafe { IoPort::new(0x0CFC) };
+use crate::device_info::PciDeviceLocation;
+
+static PCI_ADDRESS_PORT: Once<IoPort<u32, WriteOnlyAccess>> = Once::new();
+static PCI_DATA_PORT: Once<IoPort<u32, ReadWriteAccess>> = Once::new();
 
 const BIT32_ALIGN_MASK: u32 = 0xFFFC;
 
-pub(crate) fn write32(location: &PciDeviceLocation, offset: u32, value: u32) -> Result<()> {
-    PCI_ADDRESS_PORT.write(encode_as_port(location) | (offset & BIT32_ALIGN_MASK));
-    PCI_DATA_PORT.write(value.to_le());
+pub(crate) fn write32(
+    location: &PciDeviceLocation,
+    offset: u32,
+    value: u32,
+) -> Result<(), ostd::Error> {
```

Comment:
Use `Error` in `read32`/`write32` to be consistent with the non-x86 architectures.